core/src/test/java/kafka/test/api/ShareConsumerTest.java

```diff
@@ -1819,17 +1706,12 @@ private CompletableFuture<Integer> produceMessages(int messageCount) {
                 recordFutures[i] = producer.send(record);
             }
             for (int i = 0; i < messageCount; i++) {
-                try {
-                    recordFutures[i].get();
-                    messagesSent++;
-                } catch (Exception e) {
-                    fail("Failed to send record: " + e);
-                }
+                final int index = i;
+                assertDoesNotThrow(() -> recordFutures[index].get(), "Failed to send record");
```

Comment:
it  seems we expect all messages should be succeed, and hence we can rewrite this function:
```java
    private int produceMessages(int messageCount) {
        try (KafkaProducer<byte[], byte[]> producer = createProducer(new ByteArraySerializer(), new ByteArraySerializer())) {
            ProducerRecord<byte[], byte[]> record = new ProducerRecord<>(tp.topic(), tp.partition(), null, "key".getBytes(), "value".getBytes());
            IntStream.range(0, messageCount).forEach(__ -> producer.send(record));
            producer.flush();
        }
        return messageCount;
    }
```